client/client.go

```diff
@@ -89,16 +99,24 @@ func (c *Client) doWithErr(req *http.Request) (*http.Response, error) {
 
 // SendRequest verifies, constructs, and authorizes an HTTP request.
 func (c *Client) SendRequest(method string, rawURL string, data url.Values,
-	headers map[string]interface{}) (*http.Response, error) {
+	headers map[string]interface{}, body ...byte) (*http.Response, error) {
+
+	contentType := extractContentTypeHeader(headers)
+
 	u, err := url.Parse(rawURL)
 	if err != nil {
 		return nil, err
 	}
-
+	
 	valueReader := &strings.Reader{}
 	goVersion := runtime.Version()
+	var req *http.Request
 
-	if method == http.MethodGet {
+	//For HTTP GET Method there are no body parameters. All other parameters like query, path etc
+	// are added as information in the url itself. Also while Content-Type is json, we are sending 
+	// json body. In that case, data variable conatins all other parameters than body, which is the 
+	//same case as GET method. In that case as well all parameters will be added to url
+	if method == http.MethodGet || contentType == jsonContentType{
 		if data != nil {
```

Comment:
Previously only GET method was supposed to have query parameter, going forward any method can have query parameter.